racket-cas/racket-cas.rkt

```diff
@@ -996,33 +1260,67 @@
   (check-equal? (Sin (⊕ x (⊗ 3 @pi)))    (⊖ (Sin x)))
   (check-equal? (Sin (⊕ x (⊗ 2 @n @pi)))    (Sin x))
   (check-equal? (Sin (⊕ x (⊗ 4 @n @pi)))    (Sin x))
-  (check-equal? (Sin (⊕ x (⊗ 2 @p @pi)))    (Sin x)))
+  (check-equal? (Sin (⊕ x (⊗ 2 @p @pi)))    (Sin x))
+  (check-equal? (Sin (⊗ 2/3 @pi)) '(* 1/2 (expt 3 1/2)))
+  (check-equal? (Sin -3) (⊖ (Sin 3)))
+  )
+
+(require (submod "math-match.rkt" predicates))
+
+(define (terms-with-negative-coeff? s)
+  (when debugging? (displayln (list 'terms-with-negative-coeff? s)))
+  (define (list-has-negative-element? s)
+    (ormap negative-number? s))
+  (math-match (normalize s)
+              [r- #t]
+              [(Prod u) (list-has-negative-element? u)]
+              [_ #f]))
 
 (define (Asin: u)
+  (when debugging? (displayln (list 'Asin: u)))
   (math-match u
     [0 0]
     [1  (⊗ 1/2 @pi)]
-    [-1 (⊗ -1/2 @pi)]
-    ; [r (sin r)] ; nope - automatic evaluation is for exact results only
+    [1/2 (⊗ 1/6 @pi)]
+    [(list '* 1/2 (list 'expt 3 1/2))               (⊗ 1/3 @pi)]
+    [u #:when (terms-with-negative-coeff? u) (⊖ (Asin (⊖ u)))] ; odd function
     [r.0 (asin r.0)]
+    ; todo
+    ;[(Sin u) -> clamp u to [-pi/2, pi/2]
```

Comment:
maybe it is not worth it.
otherwise a new expression `Clamp` will be introduced.